File: Лаба2/Задание 1 вариант 3/main.go
package main

import (
    "fmt"
    "math"
)

func checkP(x int) int {
    var b int
    var z int
    for i := x; i > 1; i-- {
        z = 1
        for j := 2; j <= int(math.Sqrt(float64(i))); j++ {
            if i%j == 0 {
                z = 0
                break
            }
        }
        if z == 1 {
            b = i
            break
        }
    }
    for i := x; ; i++ {
        z = 1
        for j := 2; j <= int(math.Sqrt(float64(i))); j++ {
            if i%j == 0 {
                z = 0
                break
            }
        }
        if z == 1 {
            if x-b > i-x {
                b = i
            }
            break
        }
    }
    return b
}

func main() {
    var x int
    fmt.Scan(&x)
    fmt.Println(int(math.Abs(float64(x - checkP(x)))))
}

File: Лаба2/Задание 1 вариант 3/main.cpp
#include <math.h>
#include <iostream>


using namespace std;

int check_p(int x) {
    int b;
    int z;
    for (int i = x; i > 1; i--) {
        z = 1;
        for (int j = 2; j <= (int)sqrt(i); j++) {
            if (i % j == 0) {
                z = 0;
                break;
            }
        }
        if (z) {
            b = i;
            break;
        }
    }
    for (int i = x; i > 0; i++) {
        z = 1;
        for (int j = 2; j <= (int)sqrt(i); j++) {
            if (i % j == 0) {
                z = 0;
                break;
            }
        }
        if (z) {
            if (x - b > i - x) {
                b = i;
            }
            break;
        }
    }
    return b;
}

int main() {
    int x;
    cin.exceptions(istream::failbit | istream::badbit);
    try
    {
        cin >> x;
    }
    catch (istream::failure e)
    {
        cerr << "Error" << endl;
        return 1;
    }
    cout << abs(x - check_p(x)) << endl;
}
